src/rgw/rgw_op.h

```diff
@@ -331,6 +331,15 @@ class RGWGetObj : public RGWOp {
     get_legal_hold = false;
  }
 
+ protected:
```

Comment:
these data members don't belong here. probably in derived

Reply:
it must define in GetObj , since  it is been access in execute().
S3Select_OP is quite "thin" , most of its operation is done on send_response_data
the rest of the API's are re-use.

Reply:
but this really breaks the abstraction. I also think that "execute" should be overwritten in derived.
to prevent code duplication you can extract the common parts of "execute" into an internal utility function.

Reply:
its not breaking (it re-use), but it blow-up the code.
anyhow, i will remove the s3-select parts from base, the execute method will remain as before my changes.